frontend/public/components/factory/list.tsx

```diff
@@ -393,9 +399,9 @@ const stateToProps = ({UI}, {data = [], defaultSortField = 'metadata.name', defa
     if (currentSortField) {
       // Sort resources by one of their fields as a string
       sortBy = resource => sorts.string(_.get(resource, currentSortField, ''));
-    } else if (currentSortFunc && sorts[currentSortFunc]) {
+    } else if (currentSortFunc && _.has(sorts, currentSortFunc)) {
       // Sort resources by a function in the 'sorts' object
-      sortBy = sorts[currentSortFunc];
+      sortBy = _.get(sorts, currentSortFunc);
```

Comment:
This could introduce a bug if `currentSortFunc` ever has any special characters since `.` etc will be treated as path separators.

`sorts` should always be defined here, so I'd just revert the change on these two lines.

Reply:
Fixed